independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateImpl.java

```diff
@@ -231,8 +235,30 @@ protected Engine engine() {
         }
 
         private CompletionStage<String> renderAsyncNoTimeout() {
-            StringBuilder builder = new StringBuilder(1028);
-            return renderData(data(), builder::append).thenApply(v -> builder.toString());
+            StringBuilder builder = new StringBuilder(getCapacity());
+            return renderData(data(), builder::append).thenApply(v -> {
+                String str = builder.toString();
+                capacity.update(str.length());
+                return str;
+            });
+        }
+
+        private int getCapacity() {
+            if (!attributes.isEmpty()) {
+                Object c = getAttribute(TemplateInstance.CAPACITY);
```

Comment:
If `capacity::get` is the fast path, I suggest to move this part in a separate method

Reply:
Something like this?
```java
private int getCapacity() {
    if (!attributes.isEmpty()) {
      return getCapacityAttributeValue();
    } 
    return capacity.get();
 }
```